### ISSUE TITLE
H diffuse in Pd

### ISSUE THREAD

**Issue by yanxon** [yanxon/diffusion#1]
Hi @timfrol,

I hope your vacation went well.

I created this repo in an attempt to calculate the diffusivity constant.

Please check out the LAMMPS scripts.

**Comment by yanxon** [comment]
Hi @timfrol,

1. I run diffusion calculation as we discussed before. Here is the average of 100 md run with LAMMPS:
https://github.com/yanxon/diffusion/blob/master/results/100run.png

2. I also wrote my own Python script to track the hydrogen position in PBC condition. I validate the my MSD result with LAMMPS and they are identical. Here is the example for 1 md run with my script:
https://github.com/yanxon/diffusion/blob/master/results/1run.png

3. I tried to implement your suggestion by shifting the reference position (r(t=0)). I don't think this method is reliable. For example, I use the result from #2 and shift the reference point. Here is what I obtain:
![0p9](https://user-images.githubusercontent.com/32254481/61671972-a421ce80-ac9e-11e9-801a-1407183752bc.png)
Indeed, it smoothens the figure #2. But I don't think it obeys Einstein's very consistently.

**Comment by timfrol** [comment]
@yanxon 
Howard, great stuff. 
My van leaves at 5:20, not sure if there is enough time to chat today. We can certainly do it tomorrow. 

I assume that you compared your own MSQD with the one from lammps and they match perfectly. This is very good. 
I would like to look at your X, Y, Z positions plots from which you generated MSQD. 
How many times did you shift the reference points. Obviusly it is a question of statistics. You need to increase the averaging until you get the straight line. 

We should definitely chat. Looks very promising. 
Tim

**Comment by timfrol** [comment]
@yanxon 

your 1. looks pretty linear, which is nice. I would try improving 3 and compare the diffusion coefficients you get. Definitely fit 1. with a straight line. Here is an example from one of my papers.
![Screenshot 2019-07-22 16 52 33](https://user-images.githubusercontent.com/10873153/61672615-214e4300-aca1-11e9-8038-7213d842896f.png)

**Comment by yanxon** [comment]
@timfrol,

We can definitely chat tomorrow.

I shift it 1000 times. Here is where I shift 4000 times:
![0p6](https://user-images.githubusercontent.com/32254481/61672554-e77d3c80-aca0-11e9-8b33-d87df9baeccb.png)

I have the script here:
https://github.com/yanxon/diffusion/blob/master/test_script/shift.py

The script is messy. But it works. I would like to clean up so that it can be read.

**Comment by timfrol** [comment]
@yanxon 

ok, I think you got it. Do as many averaging you can with this data. I would shorten the trajectory to the time interval from 0 to 0.2. Make all the averaging you can and then supperimpose it on your plot from 1. 

Even now the section (0,0.2) begins to look pretty linear. I am wondering if it matches 1.

**Comment by yanxon** [comment]
@timfrol,

- I fit the linear equation into #1 (the title is the diffusion coefficient):
https://github.com/yanxon/diffusion/blob/master/results/fit.png
I also checked (eyeballed) my diffusion coefficient with the one in [literature](https://www.sciencedirect.com/science/article/pii/S1359646218300824). They are in the same order of magnitude.
- I also try to get the statistics in the time interval of [0, 0.2] by shifting the msd 8000 times.
https://github.com/yanxon/diffusion/blob/master/results/result.png
The D coefficient seems like twice as high.

**Comment by timfrol** [comment]
@yanxon 

sounds good Howard, do the [0,0.2] interval and let's chat after that

**Comment by timfrol** [comment]
@yanxon 

Just to clarify 

in 3d <x2>=6Dt
2d: <x2>=4Dt
1d <x2>=2Dt

When you compare the diffusion coefficient, you have to find one for the potential you are using, not the experiment or some other model.

**Comment by yanxon** [comment]
@timfrol 

In my case, I use 6Dt throughout.

[0,0.2] interval:
https://github.com/yanxon/diffusion/blob/master/results/result.png

FYI, the this paper (https://www.sciencedirect.com/science/article/pii/S1359646218300824) also using the same Potential. Therefore, I can compare.

**Comment by timfrol** [comment]
@yanxon 
Howard,

the plot looks good. Can you overlay it with the plot you got by averaging lammps MSQD over 100 separate simulations? Do you get the same number for Ds?

**Comment by timfrol** [comment]
@yanxon 

Howard,

it is good that you are looking at this paper. A good test of the methodology would be to take their system size and concentration and repeat the calculation to make sure you get the same number. 

Tim

**Comment by yanxon** [comment]
@timfrol 

I put the D coefficient in the title of the plot. Sorry if it's too small.

**Comment by yanxon** [comment]
@timfrol 

How do I tell LAMMPS that I want Pd10H4?

**Comment by yanxon** [comment]
@timfrol 

I think figure it out. So, tell if I'm wrong.
In the paper, they use 2048 Pd atoms. I just have to input ~819 hydrogen atoms. Is this correct?

**Comment by yanxon** [comment]
@timfrol 

1. I ran a NPT calculation just like the paper suggested, and used the same parameters, number of atoms, etc. The figure is the result of PdHx (x=0.4) at 600 K. D = 1853.2 A^2/ns. It's close to the result in the paper which is ~2000 A^2/ns. I only ran the MD simulation for 0.15 ns, whereas Zhou et al. ran it for 2.2 ns. I think my method is correct.
![PdH_0p4](https://user-images.githubusercontent.com/32254481/61827080-a6a83380-ae18-11e9-852d-43768c9f7633.png)

2. I also ran NVT on the same system. But, they yield very different result on the diffusion coefficients, almost 1 order of magnitude. Can you please explain why does this happen?
![PdH_0p4_NVT](https://user-images.githubusercontent.com/32254481/61828021-a5780600-ae1a-11e9-82d8-75517d5f614c.png)

**Comment by yanxon** [comment]
@timfrol 

I think the difference can be explained by pressure. With NVT simulation, the pressure is at about 20 GPa. Whereas, the NPT simulation is forcing the pressure of the system to be at 0 GPa.

**Comment by timfrol** [comment]
@yanxon 

very good Howard. You are becoming a master of MD. 
That tells me that you did not scale the simulation block correctly. Ideally, your scaled block should have had the same dimensions as the averaged dimensions in NPT. 20Gpa is a lot, you must have been off by a lot. 


Did you even confirm that MSQD reported by LAMMPS gives you the same answer as the D you get from the positions and averaging by shifting the reference time?

**Comment by timfrol** [comment]
@yanxon 

"Did you even confirm"

was a typo

I meant to write did you ever confirmed

**Comment by yanxon** [comment]
@timfrol 

I concluded that they are not the same.
![shift](https://user-images.githubusercontent.com/32254481/61838825-ef271780-ae3f-11e9-8360-9c568b5b9d89.png)

**Comment by timfrol** [comment]
@yanxon 

Howard, if you tested your MSQD from atomic positions against MSQD from LAMMPS and you get identical result, there are only two possibilities for the discrepancy 1. Something is not right about how you do the shifting and the averaging. 2. If 1. is correct then it is a question of statistics. 

The blue line is taken from a single trajectory in a single simulation. If you take a different simulation you should get a very different 'blue' line. By averaging them you will get the red line. 

The shifting the reference averaging is one of the effective methods to improve your statistics. We should figure this out before we proceed to GB simulations.

**Comment by yanxon** [comment]
@timfrol 

I just finished generating this graph:
![average](https://user-images.githubusercontent.com/32254481/62649695-25699a00-b90a-11e9-997e-6bbefff99b8f.png)
At a first glance, I think the average blue curve is populating towards the bottom of the red curve. But the average of the blue curve matches the red curve:
![average_blue](https://user-images.githubusercontent.com/32254481/62651027-f7d22000-b90c-11e9-9881-752f3cd72c2c.png)

**Comment by timfrol** [comment]
Ok, Howard,

it looks pretty good to me and I feel you understood how this averaging by shifting the reference point as well as by running MD in parallel could improve the statistics for the D calculations. I suggest that you try running larger and longer jobs on borax, so we produce the data very quickly. At which point I would calculate D for a couple of other Ts to compare with the Arrhenius plot of D from the paper. 

GBs after that and you can write your awesome paper.

**Comment by timfrol** [comment]
@yanxon 

Ok, Howard,

it looks pretty good to me and I feel you understood how this averaging by shifting the reference point as well as by running MD in parallel could improve the statistics for the D calculations. I suggest that you try running larger and longer jobs on borax, so we produce the data very quickly. At which point I would calculate D for a couple of other Ts to compare with the Arrhenius plot of D from the paper.

GBs after that and you can write your awesome paper.

**Comment by timfrol** [comment]
@yanxon 

btw, once you run some simulations on borax I would like to repeat/change your runs just to make sure all the parameters and the equilibration looks good. We can exhange the files on lc by using give yanxon1 file, take frolov2 file  commands.

**Comment by yanxon** [comment]
@timfrol 

1. I will try to do this. I will use the same parameters as mentioned in the paper and reproduce figure 1(a).

2. I install lamps on borax and quartz

3. Do you want to meet with Brandon this Friday? I will send out an email to figure out time.

**Comment by timfrol** [comment]
@yanxon 
Howard, I have been having meetings each day in the past month. I need to do some writing and I would like to spend some time on the benchmark script. I would prefer not to have a meeting for a while. It sounds like we are making progress, I would focus on work for now. Feel free to have a meeting without me. 

1. and 2. sounds great. You should have parallel lammps  and the submission script for borax  from last year right?

**Comment by timfrol** [comment]
@yanxon 
just gave you an expample of a simulation for borax. There should be a binary and the submission script

**Comment by timfrol** [comment]
@yanxon 

just type on borax

take frolov2

**Comment by yanxon** [comment]
@timfrol 

Sounds good. I think I will just postpone the meeting until I get back from the conference. I have nothing really to tell Brandon except I'm making progress on my MD training. 

I do have the submission script for borax.

Thank you, @timfrol.

I wanna be done with this grain boundary project and move on with helping you and Qiang developing the benchmark.

**Comment by timfrol** [comment]
@yanxon 

great!

**Comment by yanxon** [comment]
@timfrol 

Can you please resend Cu01.eam.alloy? I accidentally delete it.

**Comment by yanxon** [comment]
@timfrol 

Also, this script gives me error in the burn initial step part. As you suggested, we should not consider the first nanosecond of the calculation.
```
# ---------- Initialize Simulation --------------------- 
variable alat equal 3.914295757
variable xdim equal ${alat}*8
variable T equal 600
variable T2 equal ${T}
variable tstep equal 0.001
variable runtime equal 10000000
variable Srandomseed equal 2129

clear
units metal
dimension 3 
boundary p p p
atom_style atomic

# ---------- Create Atomistic Structure --------------------- 
lattice fcc ${alat}
region whole block 0 ${xdim} 0 ${xdim} 0 ${xdim} units box
create_box 2 whole
create_atoms 1 region whole
create_atoms 2 random 819 ${Srandomseed} NULL

group 1 type 1
group 2 type 2

# ---------- Define Interatomic Potential --------------------- 
pair_style	eam/alloy
pair_coeff * * PdH_zhou.eam.alloy Pd H

timestep ${tstep}

# ---------- set random velocity -----------
velocity all create ${T2} ${Srandomseed}

# Burn initial steps
fix 1 all npt temp ${T} ${T} 10 iso 0 0 10
run 10000
unfix 1

# ----- perform MD -------
fix 1 all npt temp ${T} ${T} 1 iso 0 0 1
compute msd 2 msd

#----- print ------
thermo 100 
thermo_style custom step temp c_msd[4] press
thermo_modify lost ignore flush yes

dump 1 2 custom 100 hydrogen/*.chkpt id type mass x y z

run ${runtime}

```

I just give you the potential on Borax.

**Comment by timfrol** [comment]
@yanxon 

Howard,

I forgot how to run multiple simulations from one script, but you can search it on lammps manual. you may need to split the runs by using run 0 command or reset run. I am sorry I cannot remmenber.

**Comment by timfrol** [comment]
@yanxon 

Howard,

you should use commands like

reset_timestep 0
thermo 0
run 0

to split the 2 simulations. 

On Borax, I gave you my lammps script that builds a triple junction. Doesnt matter what it is , but I run multiple simulations in that script. Take a look.

Tim

**Comment by yanxon** [comment]
@timfrol 

You're right. Due to the atom escaping, the Arrhenius plot makes no sense. I will construct the PdH system by hand. If you know happen to know a shortcut to do this, I will really appreciate it.

**Comment by timfrol** [comment]
@yanxon 

Howard, I dont think there is a shortcut with lammp, but you can write a simple script, it would be good to have it for future simulations of GBs.

**Comment by yanxon** [comment]
@timfrol 

I generated the PdH_x (x=0.4) with Python. I'm running the same input script with PdH_x, and I obtain this error:

```
ERROR on proc 0: Neighbor list overflow, boost neigh_modify one (../npair_half_bin_atomonly_newton.cpp:114)
Last command: run ${runtime}
```

Do you have any idea what's wrong? Btw, I give you the files in Borax.

I tried several method suggested from LAMMPS email such as using neigh_modify etc. Nothing seems to work.

**Comment by timfrol** [comment]
@yanxon 
Neighbor list overflow
probably means you have too many atoms in one spot. Try building PdH containing less atoms and check that it works.

**Comment by yanxon** [comment]
@timfrol 

It's running now. The dump files coordinates are in fractional coordinates. But, read_data command read the dump file in the format of cartesian coordinates.

I have to convert the fractional coordinates to cartesian coordinates. Now, it's working.

**Comment by yanxon** [comment]
@timfrol 

I have a question.
When you run npt simulation and the pressure fluctuates from -3000 to 3000 bars. Is this normal?

**Comment by timfrol** [comment]
@yanxon 

Howard,
first of all the fluctuations depend on the number of atoms. The more atoms you have smaller fluctuations.  What is your barostat constant?  there is a discussion on lammps website regarding what it should be.

**Comment by yanxon** [comment]
@timfrol 

I use 1 as the Pdamp. I tried 10, 100, 1000. It gives the same result. I have 2867 Pd and H atoms.

**Comment by timfrol** [comment]
@yanxon 

what if you try 0.1? Also this constant becomes less important for large systems. Try using about 500 to 1000 atoms per core. On borax you have 36 cores, so you can afford bigger systems, this way you get better statistics as well.

**Comment by yanxon** [comment]
@timfrol 

From what I read in the lammps documentation, the small Pdamp means more fluctuation. That's why I increase it. But, I'm trying 0.1, and it's the same.

**Comment by timfrol** [comment]
@yanxon 

I think it is ok to keep it at 1. Good that you get the same

**Comment by yanxon** [comment]
@timfrol 

I generated the Arrhenius plot:
![Figure_1](https://user-images.githubusercontent.com/32254481/64056463-3348be80-cb48-11e9-80a7-75d7313fa3dc.png)

I felt like this is wrong. Here is the 300K MD simulation:
![Screen Shot 2019-08-30 at 5 05 41 PM](https://user-images.githubusercontent.com/32254481/64056493-6ab76b00-cb48-11e9-9f4a-281fec896652.png)

400K:
![Screen Shot 2019-08-30 at 5 07 40 PM](https://user-images.githubusercontent.com/32254481/64056520-ae11d980-cb48-11e9-9ec1-f90e69ffb81b.png)

500K:
![Screen Shot 2019-08-30 at 5 10 07 PM](https://user-images.githubusercontent.com/32254481/64056571-047f1800-cb49-11e9-8753-dffc05aaec3a.png)

600K:
![Screen Shot 2019-08-30 at 5 11 38 PM](https://user-images.githubusercontent.com/32254481/64056594-3d1ef180-cb49-11e9-8538-3c2477b1f8dd.png)

**Comment by timfrol** [comment]
@yanxon 
Excellent work Howard! The Arrhenius plot looks good. 
What you have found is sometimes called the ballistic region. This is when an atom moves in a straight path without hitting anything. This is usually a shorter displacement which is different from a random walk, as such it is not expected to follow the Einstein relation. This ballistic region becomes less noticable at high T when the random walk is substantial. 

To make your low T D coefficients more accurate, people usually skip the ballistic region and only fit the large MSQDs. 

Very good work, I am glad you carefully checking your data.

I think we will be running the GB simulations in no time!

**Comment by timfrol** [comment]
@yanxon 

Howard, 
I am having second thoughts about the ballistic range. The displacement is quite large. Is it possible that your system did not equilibrate at lower temperatures?

How long is your equilibration stage?

Tim

**Comment by yanxon** [comment]
@timfrol 

The temperature equilibrate just fine. The equilibration stage is 0.1 ns. The pressure fluctuates a lot.

**Comment by timfrol** [comment]
@yanxon 

it is not the temperature or pressure only, the structure needs to equilibrate as well. 
In the past I ran 1 or 2 ns of equilibration. Lets us test the longer equilibration time at low T and see if this funny slope goes away.

**Comment by yanxon** [comment]
@timfrol 

I will try that.

**Comment by yanxon** [comment]
@timfrol 

Here is the MD for 300 K (D = 0.95 A^2/ps) after 1ns of equilibration:
![300K](https://user-images.githubusercontent.com/32254481/64072607-de3fa200-cc46-11e9-8e3a-a399bea64053.png)

400K (D = 4.18 A^2/ps):
![400K](https://user-images.githubusercontent.com/32254481/64072792-c7e71580-cc49-11e9-87ca-9e86ec3531c8.png)


Here is the new plot of Arrhenius:
![Figure_1](https://user-images.githubusercontent.com/32254481/64072798-d2091400-cc49-11e9-8651-3a136feebfd8.png)

**Comment by timfrol** [comment]
@yanxon 

excellent, we nailed that one.
It appears that the high T point doesn't follow the perfect Arrhenius behavior. Is this a question of statistics, longer simulation time? Once the slope match the published data we can move on to GB diffusion calculations.

**Comment by yanxon** [comment]
@timfrol 

I reran the MD for 2.2ns (burn 1ns) 500K and 600K. I see this kink at around 2000 ps:
![500K](https://user-images.githubusercontent.com/32254481/64089299-55823e00-ccfa-11e9-9348-be834a03d4db.png)

Perhaps, 500K behaves the same as 300K and 400 K, but it is taking longer to occur. I think.
I'm currently running longer simulation.


Are you going to be in on Monday?

**Comment by yanxon** [comment]
@timfrol 

I ran the relaxation with this script:
```
clear
units metal
dimension 3
boundary p p p
atom_style atomic

read_data PdH.dump
group 1 type 1
group 2 type 2

pair_style eam/alloy
pair_coeff * * PdH_zhou.eam.alloy Pd H

compute eng all pe/atom
compute eatoms all reduce sum c_eng

reset_timestep 0
fix 1 all box/relax iso 0.0 vmax 0.001
thermo 10
thermo_style custom step pe vol lx ly lz press pxx pyy pzz c_eatoms
min_style cg
minimize 1e-25 1e-25 5000 10000

variable natoms equal "count(all)"
variable teng equal "c_eatoms"
variable length equal "lx"
variable ecoh equal "v_teng/v_natoms"

print "Total energy (eV) = ${teng};"
print "Number of atoms = ${natoms};"
print "Lattice constant (Angstoms) = ${length};"
print "Cohesive energy (eV) = ${ecoh};"
```

I got this plot for the E(y-axis) vs V(x-axis):
![EvsV](https://user-images.githubusercontent.com/32254481/64299587-8c409a00-cf2e-11e9-955d-36e7f10bad02.png)

There might be phase transition!

**Comment by timfrol** [comment]
@yanxon 

Howard, can you please provide more details? Are you changing the number of H atoms?
Energy is a good parameter to look at. Are you saying that there are outlires? In addition it would be good to compare the initialial number of both H and Pd with the numbers after the relaxation.

**Comment by yanxon** [comment]
@timfrol 

The number of atoms stays the same, aka 2048 Pd and 819 H. Here is the log file:
```
LAMMPS (10 Aug 2015)
Reading data file ...
  orthogonal box = (0 0 0) to (33.3144 33.3144 33.3144)
  1 by 1 by 1 MPI processor grid
  reading atoms ...
  2867 atoms
2048 atoms in group 1
819 atoms in group 2
WARNING: Resetting reneighboring criteria during minimization (../min.cpp:168)
Neighbor list info ...
  1 neighbor list requests
  update every 1 steps, delay 0 steps, check yes
  master list distance cutoff = 7.35
  ghost atom cutoff = 7.35
Setting up cg style minimization ...
  Unit style: metal
Memory usage per processor = 3.65105 Mbytes
Step PotEng Volume Lx Ly Lz Press Pxx Pyy Pzz eatoms 
       0   -5342.8968    36973.849    33.314366    33.314366    33.314366     413482.7    414506.77    390085.35    435855.97   -5342.8968 
      10   -5605.3255    38094.193     33.64751     33.64751     33.64751     338194.9    339135.46    316548.03    358901.22   -5605.3255 
      20   -5823.1731    39236.944    33.980653    33.980653    33.980653    273586.35    274464.33    253496.88    292797.86   -5823.1731 
      .
      .
      .
    2440   -9803.0948    34151.414    32.444137    32.444137    32.444137  0.030968224   -61.352703   -35.198474    96.644082   -9803.0948 
Loop time of 27.5529 on 1 procs for 2440 steps with 2867 atoms

Minimization stats:
  Stopping criterion = energy tolerance
  Energy initial, next-to-last, final = 
        -5342.89683858      -9803.0947871      -9803.0947871
  Force two-norm initial, final = 28643.4 0.00342891
  Force max component initial, final = 28626.1 0.00203344
  Final line search alpha, max atom move = 0.270424 0.000549891
  Iterations, force evaluations = 2440 4190

Pair  time (%) = 26.2664 (95.3308)
Neigh time (%) = 0.171029 (0.62073)
Comm  time (%) = 0.163721 (0.594204)
Outpt time (%) = 0.020715 (0.0751826)
Other time (%) = 0.931046 (3.37912)

Nlocal:    2867 ave 2867 max 2867 min
Histogram: 1 0 0 0 0 0 0 0 0 0
Nghost:    6769 ave 6769 max 6769 min
Histogram: 1 0 0 0 0 0 0 0 0 0
Neighs:    189605 ave 189605 max 189605 min
Histogram: 1 0 0 0 0 0 0 0 0 0

Total # of neighbors = 189605
Ave neighs/atom = 66.1336
Neighbor list builds = 24
Dangerous builds = 0
Total energy (eV) = -9803.09478710058;
Number of atoms = 2867;
Lattice constant (Angstoms) = 32.4441372220902;
Cohesive energy (eV) = -3.41928663658897;
```

So, the volume will expand during the relaxation. Then at some point, perhaps, it get to a certain volume, the atoms rearrange themselves in which lowers the formation energy.

In case you want to run it yourself. Here is the PdH.dump file and the potentials:
1. https://github.com/yanxon/diffusion/blob/master/test_script/PdH.dump
2. https://github.com/yanxon/diffusion/blob/master/PdH_zhou.eam.alloy
The simulation takes very short

If you just want to run it yourself, it takes very short

**Comment by yanxon** [comment]
@timfrol 

I visualized the minimization scheme with Ovito. There is no phase transition. The minimization actually moves all the hydrogen atoms that was previously close to the Pd atoms to somewhere in between the lattice. 

Do you think it's better to run the MD simulation with the minimized structure?

**Comment by timfrol** [comment]
@yanxon 

Howard, this sounds good. Perhaps you could try inserting atoms where the minimization takes them, further from the Pd atoms into interplanar spacing. In terms of the initial configuration for the MD simulation, it should not matter too much if the atoms do not fly out. This was a test for us, it maybe more important for future GB simulations.

**Comment by yanxon** [comment]
@timfrol 

I think so too. I don't think the MD result is going to change a lot. I have been wasting time on the submission on Borax these past few days. I submitted some jobs and waited for them to run. Then, it turns out there are some bugs in the script. But, I think I have fixed it. Let's see if the MD simulation with more Temperature points works.

**Comment by yanxon** [comment]
@timfrol 

I try Borax again last night. I ran the Pd bulk simulation in parts instead of fully-automation. So, I ran 300K, 325K, 350K, ... 500K for 4 hours/submission instead of 36 hours.

After minimizing the Pd structure with 40% hydrogen, I would be able to obtain closer activation value. Now, it is about 7% error. The red circles are the data before structure minimization.
![lnDvsT](https://user-images.githubusercontent.com/32254481/65806518-b86ec580-e13e-11e9-9791-1998a55cdc30.png)

I'm gonna try to run 550K and 600K to complete the data.

**Comment by timfrol** [comment]
@yanxon 
Hi Howard,

I think the slope is good. Why do you think your curve is shifted upward? 
You are using NPT correct?

**Comment by yanxon** [comment]
@timfrol 

Both red and blue are using the same exact script (NPT). The only difference is the blue dataset ran with minimized structure. 

Perhaps, the structure is not minimized, so that the hydrogen can travel more distance before a collision. Thus, this subjected to higher diffusivity.

**Comment by timfrol** [comment]
@yanxon 

There should not be any different, perhaps your equilibration stage is too short.

**Comment by timfrol** [comment]
@yanxon 

Howard, why don't we work on this together next week. I will ask you for your scripts and I will run them myself to see where the discrepancy comes from. 

I will need your script that creates the structure with Pd and H, with and without minimization and the lammps script that runs the simulations and outputs the MSQD. In the meantime, you can focus on the potential development.

**Comment by yanxon** [comment]
Hi @timfrol ,

I give you a folder in Borax. The folder contains 6 items:
- Pd.dump: the bulk Pd dump file.
- insert_hydrogen.py: This is the script to insert hydrogen into Pd bulk.
- PdH.dump: the dump file of PdH before minimization.
- minimization.in: this is the script to minimize the PdH.dump.
- PdH.min.dump: the dump file of PdH after minimization.
- diff.in: the lammps script that run the simulation and outputs the MSQD.

When I try `give frolov2 PdH_bulk/`, it gives me this error:
```
[yanxon1@borax2:diffusion]$ give frolov2 PdH_bulk/
Giving PdH_bulk/.
/bin/cp: omitting directory 'PdH_bulk/'
Copy of PdH_bulk/ was unsuccessful.
Removing /usr/give/frolov2/yanxon1/ from give spool.
Problem running unlink.
No files given.
```

**Comment by timfrol** [comment]
@yanxon 
great, I will play with it next week

**Comment by yanxon** [comment]
@timfrol 

I created a tar.gz file so that I can give it to you on Borax. The filename is PdH_bulk.tar.gz.

Also, below is a more complete graph:
![Figure_1](https://user-images.githubusercontent.com/32254481/65993595-667fb580-e446-11e9-8fac-40ed186f0a45.png)

For higher temperature, it seems like the ln(D) agrees more with the literature. However, for the lower temperature, the ln(D) seems to have lower value.

**Comment by timfrol** [comment]
@yanxon 

Thanks Howard, I will take a look soon. Again, the biggest issue I see is that your red and blue points  are different. There should not be any difference at all. There must be a bug or something. This is the first thing to check.

**Comment by yanxon** [comment]
@timfrol 

Sounds good. Thank you, Tim.
